Zend/zend_hash.h

```diff
@@ -853,13 +853,18 @@ static zend_always_inline void *zend_hash_next_index_insert_mem(HashTable *ht, v
 {
 	zval tmp, *zv;
 
-	ZVAL_PTR(&tmp, NULL);
-	if ((zv = zend_hash_next_index_insert(ht, &tmp))) {
-		Z_PTR_P(zv) = pemalloc(size, GC_FLAGS(ht) & IS_ARRAY_PERSISTENT);
-		memcpy(Z_PTR_P(zv), pData, size);
-		return Z_PTR_P(zv);
+	void *p = pemalloc(size, GC_FLAGS(ht) & IS_ARRAY_PERSISTENT);
+	if (!p) {
```

Comment:
pemalloc can't fail because either it's using the request allocator which can't fail, or it's using the persistent allocator __zend_malloc which has a check: https://github.com/php/php-src/blob/4f450b62648952c130781ad3c58852f4a481dcc6/Zend/zend_alloc.c#L3320-L3323